services/src/datasets/create_from_workflow.rs

```diff
@@ -0,0 +1,268 @@
+pub struct RasterDatasetFromWorkflow {
+    pub name: String,
+    pub description: Option<String>,
+    pub query: RasterQueryRectangle,
+    #[schema(default = default_as_cog)]
+    #[serde(default = "default_as_cog")]
+    pub as_cog: bool,
+}
+
+/// By default, we set [`RasterDatasetFromWorkflow::as_cog`] to true to produce cloud-optmized `GeoTiff`s.
+#[inline]
+const fn default_as_cog() -> bool {
+    true
+}
+
+/// response of the dataset from workflow handler
+#[derive(Clone, Debug, Deserialize, Serialize, ToSchema)]
+pub struct RasterDatasetFromWorkflowResult {
+    pub dataset: DatasetId,
+    pub upload: UploadId,
+}
+
+impl TaskStatusInfo for RasterDatasetFromWorkflowResult {}
+
+pub struct RasterDatasetFromWorkflowTask<C: Context> {
+    pub workflow: Workflow,
+    pub session: C::Session,
+    pub ctx: Arc<C>,
+    pub info: RasterDatasetFromWorkflow,
+    pub upload: UploadId,
+    pub file_path: PathBuf,
+}
+
+impl<C: Context> RasterDatasetFromWorkflowTask<C> {
+    async fn process(&self) -> error::Result<RasterDatasetFromWorkflowResult> {
+        let operator = self.workflow.operator.clone();
+
+        let operator = operator.get_raster().context(crate::error::Operator)?;
+
+        let execution_context = self.ctx.execution_context(self.session.clone())?;
+        let initialized = operator
+            .clone()
+            .initialize(&execution_context)
+            .await
+            .context(crate::error::Operator)?;
+
+        let result_descriptor = initialized.result_descriptor();
+
+        let processor = initialized
+            .query_processor()
+            .context(crate::error::Operator)?;
+
+        let query_rect = self.info.query;
+        let query_ctx = self.ctx.query_context(self.session.clone())?;
+        let request_spatial_ref =
+            Option::<SpatialReference>::from(result_descriptor.spatial_reference)
+                .ok_or(crate::error::Error::MissingSpatialReference)?;
+        let tile_limit = None; // TODO: set a reasonable limit or make configurable?
+
+        // build the geotiff
+        let res = call_on_generic_raster_processor_gdal_types!(processor, p => raster_stream_to_geotiff(
+            &self.file_path,
+            p,
+            query_rect,
+            query_ctx,
+            GdalGeoTiffDatasetMetadata {
+                no_data_value: Default::default(), // TODO: decide how to handle the no data here
+                spatial_reference: request_spatial_ref,
+            },
+            GdalGeoTiffOptions {
+                compression_num_threads: get_config_element::<crate::util::config::Gdal>()?.compression_num_threads,
+                as_cog: self.info.as_cog,
+                force_big_tiff: false,
+            },
+            tile_limit,
+            Box::pin(futures::future::pending()), // datasets shall continue to be built in the background and not cancelled
```

Comment:
can this cause never ending tasks?

Reply:
I did not write this code - I assume this was considered when Michael wrote the abortable query execution. If it wasn't, I can think about it.